src/System.Drawing.Common/tests/mono/System.Drawing/GraphicsTests.cs

```diff
@@ -0,0 +1,3322 @@
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(7, 9).ToArgb());
+            }
+        }
+
+        // reverse, draw the fill over
+        private Bitmap DrawFillRectangle(float width)
+        {
+            Bitmap bitmap = new Bitmap(20, 20);
+            using (Graphics g = Graphics.FromImage(bitmap))
+            {
+                g.Clear(Color.Red);
+                Rectangle rect = new Rectangle(5, 5, 10, 10);
+                if (width >= 0)
+                {
+                    using (Pen pen = new Pen(Color.Blue, width))
+                    {
+                        g.DrawRectangle(pen, rect);
+                    }
+                }
+                else
+                {
+                    g.DrawRectangle(Pens.Blue, rect);
+                }
+                g.FillRectangle(Brushes.Green, rect);
+            }
+            return bitmap;
+        }
+
+        [ConditionalFact(Helpers.GdiplusIsAvailable)]
+        public void DrawFillRectangle_Width_Default()
+        {
+            // default pen size
+            using (Bitmap bitmap = DrawFillRectangle(Single.MinValue))
+            {
+                // NW - no blue border
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(4, 4).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(5, 5).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(6, 6).ToArgb());
+                // N - no blue border
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(9, 4).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(9, 5).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(9, 6).ToArgb());
+                // NE
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(16, 4).ToArgb());
+                Assert.Equal(0xFF0000FF, (uint)bitmap.GetPixel(15, 5).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(14, 6).ToArgb());
+                // E
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(16, 9).ToArgb());
+                Assert.Equal(0xFF0000FF, (uint)bitmap.GetPixel(15, 9).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(14, 9).ToArgb());
+                // SE
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(16, 16).ToArgb());
+                Assert.Equal(0xFF0000FF, (uint)bitmap.GetPixel(15, 15).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(14, 14).ToArgb());
+                // S
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(9, 16).ToArgb());
+                Assert.Equal(0xFF0000FF, (uint)bitmap.GetPixel(9, 15).ToArgb());
+                Assert.Equal(0xFF008000, (uint)bitmap.GetPixel(9, 14).ToArgb());
+                // SW
+                Assert.Equal(0xFFFF0000, (uint)bitmap.GetPixel(4, 16).ToArgb());
+                Assert.Equal(0xFF0000FF, (uint)bitmap.GetPixel(5, 15).T
```

Comment:
Might be good to rename this (and the others named like this) to capture whatever Bug 76664 was about.